--- a/canvas-api/build.gradle.kts
+++ b/canvas-api/build.gradle.kts
@@ -121,17 +_,21 @@
         java {
             srcDir(generatedDir)
             srcDir(file("../paper-api/src/main/java"))
+            srcDir(file("../canvas-api/src/main/java"))
         }
         resources {
             srcDir(file("../paper-api/src/main/resources"))
+            srcDir(file("../canvas-api/src/main/resources"))
         }
     }
     test {
         java {
             srcDir(file("../paper-api/src/test/java"))
+            srcDir(file("../canvas-api/src/test/java"))
         }
         resources {
             srcDir(file("../paper-api/src/test/resources"))
+            srcDir(file("../canvas-api/src/test/resources"))
         }
     }
 }
@@ -166,32 +_,32 @@
             suppressPomMetadataWarningsFor(it)
         }
         from(components["java"])
-        // Canvas start - add more info to the publication
+        // Canvas start - add more info to the publication // HavocFolia start - rebrand
         pom {
             name.set("canvas-api")
-            description.set("API bundle for the Canvas Minecraft server software")
-            url.set("https://github.com/CraftCanvasMC/Canvas")
+            description.set("API bundle for the HavocFolia Minecraft server software")
+            url.set("https://github.com/CraftCanvasMC/HavocFolia")
             licenses {
                 license {
                     name.set("GPL-3.0")
-                    url.set("https://github.com/CraftCanvasMC/Canvas/blob/HEAD/LICENSE")
+                    url.set("https://github.com/CraftCanvasMC/HavocFolia/blob/HEAD/LICENSE")
                     distribution.set("repo")
                 }
             }
             developers {
                 developer {
-                    id.set("canvas-team")
-                    name.set("Canvas Team")
+                    id.set("havocfolia-team")
+                    name.set("HavocFolia Team")
                     organization.set("CanvasMC")
                     organizationUrl.set("https://canvasmc.io")
                     roles.add("developer")
                 }
             }
             scm {
-                url.set("https://github.com/CraftCanvasMC/Canvas")
+                url.set("https://github.com/CraftCanvasMC/HavocFolia")
             }
         }
-        // Canvas end - add more info to the publication
+        // Canvas end - add more info to the publication // HavocFolia end - rebrand
     }
 }
 
