--- a/src/main/java/io/papermc/paper/ServerBuildInfoImpl.java
+++ b/src/main/java/io/papermc/paper/ServerBuildInfoImpl.java
@@ -41,6 +_,7 @@
     private static final String BRAND_PAPER_NAME = "Paper";
     // Canvas start - Rebrand
     private static final String BRAND_CANVAS_NAME = "Canvas";
+    private static final String BRAND_HAVOCFOLIA_NAME = "HavocFolia"; // HavocFolia - Rebrand
 
     // private static final Key BRAND_CANVAS_ID = Key.key("canvasmc", "canvas");
     // private static final Key BRAND_FOLIA_ID = Key.key("papermc", "folia");
@@ -56,9 +_,9 @@
         this(
             getManifestAttribute(manifest, ATTRIBUTE_BRAND_ID)
                 .map(Key::key)
-                .orElse(BRAND_CANVAS_ID), // Canvas - Rebrand
+                .orElse(BRAND_HAVOCFOLIA_ID), // Canvas - Rebrand // HavocFolia - Rebrand - the relevant BRAND_HAVOCFOLIA_ID sits in paper-api, check the patches there!
             getManifestAttribute(manifest, ATTRIBUTE_BRAND_NAME)
-                .orElse(BRAND_CANVAS_NAME), // Canvas - Rebrand
+                .orElse(BRAND_HAVOCFOLIA_NAME), // Canvas - Rebrand // HavocFolia - Rebrand
             SharedConstants.getCurrentVersion().id(),
             SharedConstants.getCurrentVersion().name(),
             getManifestAttribute(manifest, ATTRIBUTE_BUILD_NUMBER)
@@ -80,7 +_,7 @@
 
     @Override
     public boolean isBrandCompatible(final @NotNull Key brandId) {
-        return brandId.equals(this.brandId) || brandId.equals(BRAND_PAPER_ID) || brandId.equals(BRAND_FOLIA_ID); // Canvas - Rebrand
+        return brandId.equals(this.brandId) || brandId.equals(BRAND_PAPER_ID) || brandId.equals(BRAND_FOLIA_ID) || brandId.equals(BRAND_CANVAS_ID); // Canvas - Rebrand // HavocFolia - Rebrand
     }
 
     @Override
